flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamOperatorFactoryUtil.java

```diff
@@ -61,20 +60,14 @@
 			((ProcessingTimeServiceAware) operatorFactory).setProcessingTimeService(processingTimeService);
 		}
 
-		if (operatorFactory instanceof CoordinatedOperatorFactory) {
-			Preconditions.checkNotNull(
-					operatorEventDispatcher,
-					"The OperatorEventDispatcher should not be null.");
-			((CoordinatedOperatorFactory<OUT>) operatorFactory).setOperatorEventDispatcher(operatorEventDispatcher);
-		}
-
 		// TODO: what to do with ProcessingTimeServiceAware?
 		OP op = operatorFactory.createStreamOperator(
 			new StreamOperatorParameters<>(
 				containingTask,
 				configuration,
 				output,
-				processingTimeService));
+				processingTimeService,
+				operatorEventDispatcher));
```

Comment:
Probably can be cleaned up later. The ways we pass the parameters seem a little inconsistent here. The `YieldingOperatorFactory` and `ProcessingTimeServiceAware` both have a setter in the interface. While here we pass the dispatcher for `CoordinatedOperatorFactory`.

Reply:
I think the way `YieldingOperatorFactory` and `ProcessingTimeServiceAware` are workarounds to make this work easier without reworking the interfaces and constructors from all the pre-existing operators.

That's why I didn't want to use it for the new code. It looks much simpler to "just pass" the parameter, no?